pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -552,45 +554,78 @@ func (c *Cluster) getPVCInitContainerActivate(mountPath string, osdProps osdProp
 	}
 }
 
-func (c *Cluster) getPVCEncryptionOpenInitContainerActivate(osdProps osdProperties) v1.Container {
-	container := v1.Container{
-		Name:  blockEncryptionOpenInitContainer,
+func (c *Cluster) generateEncryptionOpenBlockContainer(resources v1.ResourceRequirements, containerName, pvcName string) v1.Container {
+	return v1.Container{
+		Name:  containerName,
 		Image: c.spec.CephVersion.Image,
 		Command: []string{
 			"cryptsetup",
 		},
 		// Runs "cryptsetup --verbose --key-file /etc/ceph/luks_key --allow-discards luksOpen /set1-data-0-7dwll set1-data-0-7dwll-block-dmcrypt"
-		Args: []string{"--verbose", "--key-file", encryptionKeyPath(), "--allow-discards", "luksOpen", fmt.Sprintf("/%s", osdProps.pvc.ClaimName), encryptionDMName(osdProps.pvc.ClaimName)},
+		Args: []string{
+			"--verbose",
+			"--key-file",
+			encryptionKeyPath(),
+			"--allow-discards",
+			"luksOpen",
+			fmt.Sprintf("/%s", pvcName),
+			encryptionDMName(pvcName),
+		},
 		VolumeDevices: []v1.VolumeDevice{
 			{
-				Name:       osdProps.pvc.ClaimName,
-				DevicePath: fmt.Sprintf("/%s", osdProps.pvc.ClaimName),
+				Name:       pvcName,
+				DevicePath: fmt.Sprintf("/%s", pvcName),
 			},
 		},
 		VolumeMounts:    []v1.VolumeMount{getDeviceMapperMount()},
 		SecurityContext: opmon.PodSecurityContext(),
-		Resources:       osdProps.resources,
+		Resources:       resources,
 	}
+}
+
+func (c *Cluster) getPVCEncryptionOpenInitContainerActivate(osdProps osdProperties) []v1.Container {
+	containers := []v1.Container{}
+	containers = append(containers, c.generateEncryptionOpenBlockContainer(osdProps.resources, blockEncryptionOpenInitContainer, osdProps.pvc.ClaimName))
```

Comment:
nit: initialize the container with all its properties to a local var, then append it to the containers slice. This way, you wouldn't have to reference the slice with `containers[0]` or `containers[1]` below. It's more readable IMO that way.